### PR TITLE
Added SOAP, required by eZ Commerce

### PR DESCRIPTION
**Pull request by jheba**
eZ Commerce need a SOAP to fetch prices of the products.
@andrerom and @vidarl please wash the PR ;-)

### CODE REVIEW

File: php/Dockerfile-7.0 [hunk 1]
```diff
@@ -26,6 +26,7 @@ RUN apt-get update -q -y \
         libmemcachedutil2 \
 # Required by eZ Commerce
         imagemagick \
+        libxml2-dev \
```
Comment (andrerom) [inline php/Dockerfile-7.0]: build deps we usually put in `buildDeps`
Is there some more specific package you need after compilation + this one in buildDeps perhaps?

File: php/Dockerfile-7.3 [hunk 2]
```diff
@@ -27,6 +27,7 @@ RUN apt-get update -q -y \
         libzip4 \
 # Required by eZ Commerce
         imagemagick \
+        libxml2-dev \
```
Comment (andrerom) [inline php/Dockerfile-7.3]: Nitpick, but to stay consistent with the convention here you should rather install libxml2 into image, and `libxml2-dev` only during build in `buildDeps` section:

```diff
@@ -27,7 +27,7 @@ RUN apt-get update -q -y \
        libzip4 \
# Required by eZ Commerce
        imagemagick \
-      libxml2-dev \
+     libxml2 \
# git & unzip needed for composer, unless we document to use dev image for composer install
# unzip needed due to https://github.com/composer/composer/issues/4471
        unzip \
@@ -44,6 +44,7 @@ RUN set -xe \
        libpng-dev \
        libicu-dev \
        libxslt1-dev \
+	libxml2-dev \
        libmemcached-dev \
        libzip-dev \
    " \
```

File: php/Dockerfile-7.0 (hunk 1 above)
Comment (jheba) [reply php/Dockerfile-7.0]: Thanks. Indeed the image becomes much smaller with this approach.
Corrected.

**Review by andrerom — APPROVED** [review]
+1

I realize libxml2 (non dev) is implicit already installed by `libxslt1.1` so this looks ok.
And failure seems random so new CI job should hopefully pass.

File: php/Dockerfile-5.6 [hunk 3]
```diff
@@ -24,6 +24,8 @@ RUN apt-get update -q -y \
         libicu52 \
         libxslt1.1 \
         libmemcachedutil2 \
+# Required by eZ Commerce (added here to keep images in sync, feature wise)
```
Comment (vidarl) [inline php/Dockerfile-5.6]: eZ Commerce do not run on php 5.6....

File: php/Dockerfile-7.0 [hunk 4]
```diff
@@ -24,7 +24,7 @@ RUN apt-get update -q -y \
         libicu52 \
         libxslt1.1 \
         libmemcachedutil2 \
-# Required by eZ Commerce
+# Required by eZ Commerce (added here to keep images in sync, feature wise)
```
Comment (vidarl) [inline php/Dockerfile-7.0]: If such comments (required by eZ commerce) is needed, that should be added to the soap line below to..
IMO it is not needed though ( reason is anyway in gitlog )

**Review by vidarl — CHANGES_REQUESTED** [review]

File: php/Dockerfile-5.6 (hunk 3 above)
Comment (andrerom) [reply php/Dockerfile-5.6]: So these are my changes, as I'd like images to have the same feature set and be simple to keep in sync. hence the `(added here to keep images in sync, feature wise)` to answer comments like:
> eZ Commerce do not run on php 5.6....

If you think it's still wrong we can drop it, but..

Comment (vidarl) [reply php/Dockerfile-5.6]: Ok. I don't mind having it there then if it simplifies things
My point was just that eZ Platform 2.3 [requires php 7.1 or newer](https://github.com/ezsystems/ezplatform/blob/2.3/composer.json#L27)

Comment (andrerom) [reply php/Dockerfile-5.6]: Yes I know, but these images are also for 1.x, and would like to:
- simplify maintaining them (so keeping them as much as possible in sync)
- have similar feature set / behaviour when moving form one php version to the other, e.g. platform can use imagemagic also, it's just that commerce has hardcoded this and for now does not support GD.

File: php/Dockerfile-7.0 (hunk 4 above)
Comment (jheba) [reply php/Dockerfile-7.0]: @vidarl I removed your comments about eZ Commerce from all the Dockerfiles here.
Could you re-review the PR?

**Review by vidarl — APPROVED** [review]

File: php/Dockerfile-7.0 (hunk 4 above)
Comment (vidarl) [reply php/Dockerfile-7.0]: done

**Comment by andrerom** [comment]
Awaiting travis before merge.

**Pull request merged**